### DISCUSSION TITLE
Test does not compile when`Map[Long, String]` compare to `Map[Int, String]`.

### DISCUSSION THREAD

**Discussion by jxnu-liguobin** [zio/zio#8177] [Q&A]
<img width="792" alt="企业微信截图_c134f5df-0acf-4683-8a92-fdebd712c862" src="https://github.com/zio/zio/assets/29496873/459c5f01-a04d-4516-8cc7-0233dc39c9eb">
<img width="1528" alt="image" src="https://github.com/zio/zio/assets/29496873/3236dbab-628a-4df6-b69a-852bb1d9131f">

The test is pass when the types are all Long.
<img width="1106" alt="image" src="https://github.com/zio/zio/assets/29496873/bad3f10d-ad2a-4486-84bb-cf15597cf124">

Scala
<img width="363" alt="image" src="https://github.com/zio/zio/assets/29496873/24423f1d-d99c-4808-bac5-7ba32aba2173">

**Comment by adamgfraser** [comment]
I think this is expected behavior. `Int` and `Long` are unrelated types. Of course they are both numeric types but that kind of implicit numeric widening is normally regarded as an anti-pattern.
